packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,369 @@
+              verticalController: _verticalController,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _LoggingTableContents extends StatefulWidget {
+  _LoggingTableContents({
+    required this.model,
+    required ScrollController verticalController,
+  }) : _verticalController = verticalController;
+
+  static const _millisecondsUntilCacheProgressShows = 500;
+  static const _millisecondsUntilCacheProgressHelperShows = 2000;
+
+  final LoggingTableModel model;
+  final ScrollController _verticalController;
+
+  @override
+  State<_LoggingTableContents> createState() => _LoggingTableContentsState();
+}
+
+class _LoggingTableContentsState extends State<_LoggingTableContents> {
+  final Stopwatch _progressStopwatch = Stopwatch();
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        widget.model.tableWidth = constraints.maxWidth;
+        _progressStopwatch.reset();
+        return ListenableBuilder(
+          listenable: widget.model.cacheLoadProgress,
+          builder: (context, _) {
+            final cacheLoadProgress = widget.model.cacheLoadProgress.value;
+            if (cacheLoadProgress != null) {
+              double progress = cacheLoadProgress;
+              if (!_progressStopwatch.isRunning) {
+                _progressStopwatch.start();
+              }
+
+              if (_progressStopwatch.elapsedMilliseconds <
+                  _LoggingTableContents._millisecondsUntilCacheProgressShows) {
+                progress = 0.0;
+              }
+
+              return Center(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    const Text(
+                      'Resizing... this will only take a moment.',
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    SizedBox(
+                      width: defaultLinearProgressIndicatorWidth,
+                      height: defaultLinearProgressIndicatorHeight,
+                      child: LinearProgressIndicator(
+                        value: progress,
+                      ),
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    if (_progressStopwatch.elapsedMilliseconds >
+                        _LoggingTableContents
+                            ._millisecondsUntilCacheProgressHelperShows)
+                      const Text(
+                        'To make this process faster, then reduce the "Log Retention" setting.',
+                      ),
+                  ],
+                ),
+              );
+            } else {
+              _progressStopwatch.stop();
+              _progressStopwatch.reset();
+              return Scrollbar(
+                thumbVisibility: true,
+                controller: widget._verticalController,
+                child: ListenableBuilder(
```

Comment:
does the `SliverVariedExtentList` really only build the rows it needs? or does it build everything? Having the list be defined below the listenable builder makes me suspicious.

Reply:
I'm not sure about best practices so I switched to using setState. Is that the better way to go about these things?

I can also do some testing if there is further doubt, but I remember in my demo that it was only loading a limited number of widgets at any time.

Reply:
Note: look into whether the cache around the window needs to be shrunk.